File: 조준장/level2/가장 큰 정사각형 찾기.py
def solution(board):
    answer = 0
    
    for i in range(len(board)):
        for j in range(len(board[0])):
            if board[i][j] == 1:
                board[i][j] = min(board[i-1][j], board[i][j-1], board[i-1][j-1]) + 1
    
    for k in board:
        answer = max(answer, max(k))
    
    return answer**2

File: 이훈일/Programmers/Level2/가장 큰 정사각형 찾기.java
class Solution{
    public static int solution(int [][]board) {
        int[][] dp = new int[board.length + 1][board[0].length + 1];
        for(int i = 0; i < board.length; i++) {
            for(int j = 0; j < board[i].length; j++) {
                dp[i+1][j+1] = board[i][j];
            }
        }
        
        int max = 0;
        for(int i = 1; i < dp.length; i++) {
            for(int j = 1; j < dp[i].length; j++) {
                if(dp[i][j] == 1) {
                    int left = dp[i][j-1];
                    int up = dp[i-1][j];
                    int diagonal = dp[i-1][j-1];
                    
                    dp[i][j] = Math.min(left, Math.min(up, diagonal)) + 1;
                    
                    max = Math.max(max, dp[i][j]);
                }
            }
        }
        
        return max * max;
    }
}
